native/libp2p_port/internal/discovery/discovery.go

```diff
@@ -0,0 +1,103 @@
+package discovery
+
+import (
+	"crypto/rand"
+	"fmt"
+	"libp2p_port/internal/port"
+	"libp2p_port/internal/proto_helpers"
+	"libp2p_port/internal/reqresp"
+	"libp2p_port/internal/utils"
+	"net"
+
+	"github.com/ethereum/go-ethereum/p2p/discover"
+	"github.com/ethereum/go-ethereum/p2p/enode"
+	"github.com/ethereum/go-ethereum/p2p/enr"
+	"github.com/libp2p/go-libp2p/core/crypto"
+	"github.com/libp2p/go-libp2p/core/peer"
+)
+
+type Discoverer struct {
+	port           *port.Port
+	discv5_service *discover.UDPv5
+}
+
+func NewDiscoverer(p *port.Port, listener *reqresp.Listener, config *proto_helpers.Config) Discoverer {
+	if !config.UseDiscv5 {
```

Comment:
Does this mean we only support discv5? If that's the case we should consider:
1) Documenting that.
2) Why do we need a boolean if that's the case? Are we planning on supporting something else?

Reply:
Oh, if that's the case:
1) I'd call it as you suggest, which is clearer.
2) I'd remove that logic from here. It's strange to even build the struct if we don't enable discovery. I'd move this if to `main`. What do you think?